extensions/modules/src/org/exist/xquery/modules/expathrepo/InstallFunction.java

```diff
@@ -150,15 +149,25 @@ private URI _getURI(String s) throws XPathException
         }
     }
 
-    private BinaryDocument _getDocument(String path) throws XPathException {
-    	try {
-			XmldbURI uri = XmldbURI.createInternal(path);
-			DocumentImpl doc = context.getBroker().getXMLResource(uri, LockMode.READ_LOCK);
-			if (doc.getResourceType() != DocumentImpl.BINARY_FILE)
-				throw new XPathException(this, EXPathErrorCode.EXPDY001, path + " is not a valid .xar", new StringValue(path));
-			return (BinaryDocument) doc;
-		} catch (PermissionDeniedException e) {
-			throw new XPathException(this, EXPathErrorCode.EXPDY003, e.getMessage(), new StringValue(path), e);
-		}
+  private BinaryDocument _getDocument(String path) throws XPathException {
+    try {
+      XmldbURI uri = XmldbURI.createInternal(path);
+      DocumentImpl doc = context.getBroker().getXMLResource(uri, LockMode.READ_LOCK);
+      if (doc == null) {
+        throw new XPathException(this, EXPathErrorCode.EXPDY001,
+            path + " is no .xar resource",
+            new StringValue(path)
+        );
+      } else if (doc.getResourceType() != DocumentImpl.BINARY_FILE) {
+        doc.getUpdateLock().release(LockMode.READ_LOCK);
+        throw new XPathException(this, EXPathErrorCode.EXPDY001,
+            path + " is not a valid .xar, it's not a binary resource",
```

Comment:
" is not a valid .xar, it's not a binary resource" -> " is not a valid .xar EXPath Package"